bip322/src/bitcoin_minimal.rs

```diff
@@ -0,0 +1,1278 @@
+                Ok(ScriptBuf { inner: script })
+            }
+            AddressType::P2WPKH => {
+                // P2WPKH script: OP_0 <20-byte-pubkey-hash>
+                let pubkey_hash = self.pubkey_hash.ok_or(AddressError::MissingRequiredData)?;
+                let mut script = Vec::new();
+                script.push(0x00); // OP_0
+                script.push(20); // Push 20 bytes
+                script.extend_from_slice(&pubkey_hash);
+                Ok(ScriptBuf { inner: script })
+            }
+            AddressType::P2WSH => {
+                // P2WSH script: OP_0 <32-byte-script-hash>
+                let witness_program = self
+                    .witness_program
+                    .as_ref()
+                    .ok_or(AddressError::MissingRequiredData)?;
+
+                if witness_program.program.len() != 32 {
+                    return Err(AddressError::InvalidWitnessProgram);
+                }
+
+                let mut script = Vec::new();
+                script.push(0x00); // OP_0
+                script.push(32); // Push 32 bytes
+                script.extend_from_slice(&witness_program.program);
+                Ok(ScriptBuf { inner: script })
+            }
+        }
+    }
+}
+
+/// Implementation of address parsing from the string format.
+///
+/// This implementation supports parsing the two most common Bitcoin address formats
+/// with full validation including checksum verification.
+impl std::str::FromStr for Address {
+    type Err = AddressError;
+
+    /// Parses a Bitcoin address string into an `Address` structure.
+    ///
+    /// This method performs comprehensive validation including
+    /// - Format detection (P2PKH, P2SH, P2WPKH, P2WSH)
+    /// - Encoding validation (`Base58Check` vs Bech32)
+    /// - Checksum verification
+    /// - Length validation
+    /// - Network validation (mainnet only)
+    ///
+    /// # Arguments
+    ///
+    /// * `s` - The address string to parse
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(Address)` if parsing succeeds with valid checksum
+    /// - `Err(AddressError)` if parsing fails for any reason
+    ///
+    /// # Examples
+    ///
+    /// ```rust,ignore
+    /// let p2pkh: Address = "1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa".parse()?;
+    /// let p2sh: Address = "3EktnHQD7RiAE6uzMj2ZifT9YgRrkSgzQX".parse()?;
+    /// let p2wpkh: Address = "bc1q9vza2e8x573nczrlzms0wvx3gsqjx7vavgkx0l".parse()?;
+    /// let p2wsh: Address = "bc1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3qccfmv3".parse()?;
+    /// ```
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // P2PKH (Pay-to-Public-Key-Hash) address parsing
+        // These are legacy Bitcoin addresses starting with '1' on the mainnet
+        if s.starts_with('1') {
+            // Decode the Base58Check encoded address
+            // Base58Check = Base58(version + payload + checksum)
+            let decoded = bs58::decode(s)
+                .into_vec()
```

Comment:
you expect `[u8; 25]`, so you can use https://docs.rs/bs58/latest/bs58/decode/struct.DecodeBuilder.html#method.into_array_const